test/integration/Android/FiltersTests.cs

```diff
@@ -0,0 +1,68 @@
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//See the NOTICE file distributed with this work for additional
+//information regarding copyright ownership.
+//You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+using System.Collections.Generic;
+using NUnit.Framework;
+using OpenQA.Selenium.Appium.Android;
+using OpenQA.Selenium.Appium;
+
+namespace Appium.Net.Integration.Tests.helpers
+{
+    [TestFixture]
+    public class FiltersTests
+    {
+        private AndroidDriver _driver;
+
+        [OneTimeSetUp]
+        public void BeforeAll()
+        {
+            var capabilities = Env.ServerIsRemote()
+                ? Caps.GetAndroidUIAutomatorCaps(Apps.Get("androidApiDemos"))
+                : Caps.GetAndroidUIAutomatorCaps(Apps.Get("androidApiDemos"));
+            var serverUri = Env.ServerIsRemote() ? AppiumServers.RemoteServerUri : AppiumServers.LocalServiceUri;
+            _driver = new AndroidDriver(serverUri, capabilities, Env.InitTimeoutSec);
+            _driver.Manage().Timeouts().ImplicitWait = Env.ImplicitTimeoutSec;
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _driver.StartActivity("io.appium.android.apis", ".ApiDemos");
+        }
+
+        [Test]
+        public void FirstWithName_ReturnsCorrectElement()
+        {
+            var viewsElement = _driver.FindElement(MobileBy.AccessibilityId("Views"));
+
+            var textElemet = _driver.FindElement(MobileBy.AccessibilityId("Text"));
+
+            var elements = new List<AppiumElement> { viewsElement, textElemet };
+
+            var result = Filters.FirstWithName(elements, "Text");
```

Comment:
using this API might be slow and ineffective especially on lengthy lists. The reason is that it fetches the requested attribute for each list item via server APIs. Normally, you'd need to reduce the amount of client-server calls to the very minimum though. 

For example, in the above example it would be much faster to use an appropriate XPath matcher

Reply:
I'm not sure I understand. This test aims to check the functionality of the `FirstWithName` method.
If I use the XPATH matcher, we won't use the appropriate method.

Reply:
yes, I get it
just wanted to mention such approach would not be very effective to use in general

Reply:
I'm pretty sure No one is using this method since it's under integration.Tests helpers.